tests/test_process.py

```diff
@@ -90,6 +91,248 @@ def test_upload_dataversion_metadata_without_team_images_id(self, syn: Any):
+            return df.assign(new_key=test_threshold)
+
+        return _mock_transform_with_args
+
+    @pytest.mark.parametrize(
+        "example_config_with_custom_transform,expectation",
+        # Fails because mock_transform_function_false doesn't exist
+        [
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": {
+                            "mock_transform_function_false": {
+                                "test_threshold": 1,
+                                "test_p_value": 2,
+                            }
+                        },
+                    }
+                },
+                pytest.raises(AttributeError),
+            ),
+            # Fails because mock_transform_function_false doesn't exist
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": "mock_transform_function_false",
+                    },
+                },
+                pytest.raises(AttributeError),
+            ),
+            # Fails because mock_transform_function_false is mapped to an integer
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": 1,
+                    },
+                },
+                pytest.raises(AttributeError),
+            ),
+        ],
+        ids=[
+            "invalid_custom_transformations_with_special_params",
+            "invalid_custom_transformations_with_standard_params",
+            "invalid_type",
+        ],
+    )
+    def test_apply_invalid_custom_transformations(
+        self,
+        example_config_with_custom_transform: Dict[str, Any],
+        expectation: Exception,
+    ) -> None:
+        """Test that invalid custom transformations raise an error."""
+        with expectation:
```

Comment:
This is a little confusing, I would use a more descriptive variable name like `exception` or `error_raised`

Reply:
Sure. I can change that to `expected_error`